cmd/apply.go

```diff
@@ -0,0 +1,482 @@
+	Cmd       string          `json:"cmd,omitempty" yaml:"cmd,omitempty"`
+	Arguments CommandArgument `json:"arguments,omitempty" yaml:"arguments,omitempty"`
+}
+
+type CommandArgument struct {
+	Name string `json:"name,omitempty" yaml:"name,omitempty"`
+	Type string `json:"type,omitempty" yaml:"type,omitempty"`
+}
+
+type Settings struct {
+	Enabled bool `json:"enabled,omitempty" yaml:"enabled,omitempty"`
+}
+
+type WorkflowConfig struct {
+	Name string `json:"name,omitempty" yaml:"name,omitempty"`
+	Type string `json:"type,omitempty" yaml:"type,omitempty"`
+}
+
+func ApplyAdvancedTaskDefinitions(lc *client.Client, fileConfig *FileConfigRoot) error {
+	var advancedTaskDefinitionResult *schema.AdvancedTaskDefinition
+	var data []output.Data
+
+	// Apply from file
+	hasTasks := len(fileConfig.Task)
+
+	// Add task definition for each task found
+	if hasTasks > 0 {
+		for _, t := range fileConfig.Task {
+			var hasTaskMatches = false
+
+			// If project or environment arguments are given, use those.
+			if cmdProjectName != "" {
+				t.Project = cmdProjectName
+			}
+			if cmdProjectEnvironment != "" {
+				t.Environment = cmdProjectEnvironment
+			}
+
+			// Required input checks
+			if t.Name == "" {
+				fmt.Println("Task name is required")
+				os.Exit(1)
+			}
+			if t.Project == "" {
+				fmt.Println("Project name is required")
+				os.Exit(1)
+			}
+			if t.Environment == "" {
+				fmt.Println("An Environment name is required")
+				os.Exit(1)
+			}
+
+			// Get project environments from name
+			project, err := lagoon.GetMinimalProjectByName(context.TODO(), t.Project, lc)
+			if err != nil {
+				return err
+			}
+
+			// Find environment ID if given
+			var envID int
+			if len(project.Environments) > 0 && t.Environment != "" {
+				for _, e := range project.Environments {
+					if e.Name == t.Environment {
+						envID = int(e.ID)
+					}
+				}
+			}
+
+			// Get current tasks by Environment ID
+			liveAdvancedTasks, err := lagoon.GetAdvancedTasksByEnvironment(context.TODO(), envID, lc)
+			if err != nil {
+				return err
+			}
+
+			// Check if given task already exists
+			if liveAdvancedTasks != nil {
+				for _, currAdvTask := range *liveAdvancedTasks {
+					// Convert AdvancedTaskDefinition struct from file to input the update mutation can understand
+					advancedTaskInput := &schema.AdvancedTaskDefinitionInput{
+						ID:          currAdvTask.ID,
+						Name:        t.Name,
+						Description: t.Description,
+						Type:        t.Type,
+						Service:     t.Service,
+						Image:       t.Image,
+						Command:     t.Command,
+						GroupName:   t.GroupName,
+						Project:     currAdvTask.Project,
+						Environment: currAdvTask.Environment,
+						Permission:  t.Permission,
+					}
+
+					// Marshal to JSON to flip capitalisation of struct keys from yaml unmarshalling
+					encodedJSONTask, err := json.Marshal(advancedTaskInput)
+					if err != nil {
+						fmt.Errorf("Unable to marhsal yaml config to JSON '%s': %v", t, err)
```

Comment:
This line errors, you need to return the error
```
return fmt.Errorf("Unable to marhsal yaml config to JSON '%v': %v", t, err)
```

Also the %s is for string, where the data for `t` isn't. So you need to either use %v to just print the value, or figure out what in the struct should be returned in the error.